k4GeneratorsConfig/src/key4HEPAnalysis.cxx

```diff
@@ -120,39 +120,34 @@ int main(int argc, char** argv)
   for (size_t i = 0; i < reader->getEntries(podio::Category::Event); ++i) {
     if (i)
       event = podio::Frame(reader->readNextEntry(podio::Category::Event));
-    auto& mcParticles = event.get<edm4hep::MCParticleCollection>(edm4hep::labels::MCParticles);
+    const auto& mcParticles = event.get<edm4hep::MCParticleCollection>(edm4hep::labels::MCParticles);
     // do more stuff with this event
-    edm4hep::LorentzVectorM *particleA;
-    edm4hep::LorentzVectorM *particleB;
+    edm4hep::LorentzVectorM *particleA = nullptr;
+    edm4hep::LorentzVectorM *particleB = nullptr;
     for (auto part: mcParticles){
       if ( part.getPDG() == particlesList[0] && !particleA ){
-	auto momentumA = part.getMomentum();
-	particleA = new edm4hep::LorentzVectorM(momentumA.x, momentumA.y, momentumA.z, part.getMass());
-	double costheta = cos(particleA->Theta());
-	pdgAcostheta->Fill(costheta);
+        auto momentumA = part.getMomentum();
+        particleA = new edm4hep::LorentzVectorM(momentumA.x, momentumA.y, momentumA.z, part.getMass());
+        double costheta = cos(particleA->Theta());
+        pdgAcostheta->Fill(costheta);
       }
       if ( part.getPDG() == particlesList[1] && !particleB ){
-	auto momentumB = part.getMomentum();
-	particleB = new edm4hep::LorentzVectorM(momentumB.x, momentumB.y, momentumB.z, part.getMass());
-	double costheta = cos(particleB->Theta());
-	pdgBcostheta->Fill(costheta);
+        auto momentumB = part.getMomentum();
+        particleB = new edm4hep::LorentzVectorM(momentumB.x, momentumB.y, momentumB.z, part.getMass());
+        double costheta = cos(particleB->Theta());
+        pdgBcostheta->Fill(costheta);
       }
       if ( particleA && particleB ){
-	edm4hep::LorentzVectorM part = *particleA + *particleB;
-	mpdgapdgb->Fill(part.mass());
-	ptpdgapdgb->Fill(part.pt());
-	pzpdgapdgb->Fill(part.pz());
+        edm4hep::LorentzVectorM pp = *particleA + *particleB;
+        mpdgapdgb->Fill(pp.mass());
+        ptpdgapdgb->Fill(pp.pt());
+        pzpdgapdgb->Fill(pp.pz());
       }
     }
+
     // release memory after an event
-    if ( particleA ) {
```

Comment:
With `nullptr` add the beginning, it should be possible just delete the two particles.